cores/esp8266/uart.c

```diff
@@ -452,8 +481,10 @@ uart_flush(uart_t* uart)
     if(uart->tx_enabled)
         tmp |= (1 << UCTXRST);
 
-    USC0(uart->uart_nr) |= (tmp);
-    USC0(uart->uart_nr) &= ~(tmp);
+    if(!gdbstub_has_uart_isr_control() || uart->uart_nr != UART0) {
```

Comment:
it can be read as `if (!( gdbstub_set_uart_isr_callback() && uart->uart_nr == UART0 )) do-this"
= "do this if gdb hasn't control over uart0".
It seems OK.